Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -590,10 +592,38 @@ private class TopSitesDataSource: NSObject, UICollectionViewDataSource {
         self.historySites = historySites
     }
 
-    private func filterSuggestedSites() {
+    private func filterAndMergeSuggestedSites() {
+        // We need to filter from the complete suggested sites list every time.
+        // This is needed for edge cases, such as, clearing private data
+        // with a deduped topsite will not display the suggested site again
+        // unless you reload the Top Sites panel.
+        suggestedSites = SuggestedSites.asArray()
+
         for url in profile.prefs.arrayForKey("topSites.deletedSuggestedSites") as! [String] {
-            let domainURL = NSURL(string: url)?.normalizedHost() ?? url
-            suggestedSites = SuggestedSites.asArray().filter({ $0.tileURL.normalizedHost()?.URLString != domainURL.URLString })
+            let domainURL = extractDomainURL(url)
+            suggestedSites = suggestedSites.filter({ $0.tileURL.normalizedHost()?.URLString != domainURL.URLString })
+        }
+
+        var i = 0
+        while i < self.historySites.count {
```

Comment:
What in tarnation is this?! Certainly you should know how to write a `for` loop in Swift by now...